### PR TITLE
Rename asm.jmpsub to asm.sub.jmp

### PR DESCRIPTION
**Pull request by officialcjunior**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

On #16801, it was proposed to organize `asm~sub` to `asm.sub.*`. 
Here, I've renamed `asm.jmpsub` to `asm.sub.jmp` and renamed the node `jmpsub` to `subjmp`.

**Test plan**
A simple build from source will update the changes.

**Closing issues**

Related to #16801 

(Being a beginner, I was going through the 'good first issues'. Any forms of advices and opinions are welcome on this, as I'd like to work on the whole issue and remaining commands under this.)

### CODE REVIEW

**Comment by XVilka** [comment]
You need to change the tests, man pages, help, and [radare2 book](https://github.com/radareorg/radare2book) also.

**Comment by trufae** [comment]
There are many jmp. vars under anal., but none in asm. 

there's asm.var.sub

```
[0x00000000]> e asm.~sub
asm.jmpsub = true
asm.regsub = false
asm.relsub = true
asm.section.sub = false
asm.tailsub = false
asm.var.sub = true
asm.var.submin = 256
asm.var.subonly = true
```

i would say that we probably want to have asm.sub.{rel,jmp,reg,tail,var} .. about var im not sure but it would make sense to be consistent. so i think we need to at least change in the same pr the other asm.*sub variables (rel,reg,jmp,tail)

**Comment by officialcjunior** [comment]
> You need to change the tests, man pages, help, and [radare2 book](https://github.com/radareorg/radare2book) also.

By 'help', do you mean the descriptions that show up under each commands on `Ve` ? 
When completed, to where should the descriptions be added on the r2book?

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -406,10 +406,10 @@ static bool cb_asmvarsubmin(void *user, void *data) {
 	return true;
 }
 
-static bool cb_asmtailsub(void *user, void *data) {
+static bool cb_asmsubtail (void *user, void *data) {
```
Comment (trufae) [inline libr/core/cconfig.c:409]: remove this space before the (

Comment (officialcjunior) [reply libr/core/cconfig.c]: Sorry!
Fixed that.

**Comment by officialcjunior** [comment]
Thanks. 
I hope either one of you can rename this to an appropriate name, as this affects `asm.rel`, `reg` , `jmp` and `tail` now.

**Comment by trufae** [comment]
Yes @ret2libc i think those should be renamed too

**Comment by ret2libc** [comment]
Thanks!

**Pull request merged**